compile_state/src/state.rs

```diff
@@ -1306,6 +1458,56 @@ impl SloshVmTrait for SloshVm {
     }
 
     fn global_intern_slot(&self, symbol: Interned) -> Option<u32> {
+        fn check_global(vm: &SloshVm, ns: &str, sym: &str) -> Option<u32> {
+            let mut ns = ns.to_string();
+            ns.push_str("::");
```

Comment:
can namespaces be nested? e.g.

`reptar_lib::strings::misc`?

Reply:
Not yet but I want to allow that.  It will require a variation on with-ns, currently with-ns saves the current ns, clears it and restores it when done.  Building a version that "extended" the current namespace would trivial and I will add that soon.